extended-history-analyzer.py

```diff
@@ -649,10 +654,14 @@ def create_top_n_chart_by_playtime(
         logging.basicConfig(level=logging.INFO, format="%(message)s")
 
     # Configuration
-    file_pattern = "data/chris/Streaming_History_Audio_*[0-9].json"  # Path to json files
+    file_pattern = f"data/{args.user}/Streaming_History_Audio_*[0-9].json"  # Path to json files
     output_file = "spotify_analysis_output.txt"
     target_artists = ["Coldplay"]
 
+    if not os.path.exists(file_pattern):
```

Comment:
This conditional seems to check for a file at `file_pattern`. But because `file_pattern` is meant as a shell-style pattern to be used by glob.glob(), no exact file will exist and the conditional will incorrectly evaluate to False.
```suggestion
    if not glob.glob(file_pattern):
```